status: 2023q2: mfsBSD building integration

soobinrho opened the pull request:
This is a 2023Q2 status report for the ***Integrate mfsBSD into the release building tools*** project.

I used the status report template, and made sure to follow the guidelines -- e.g. one sentence per line. Thanks everyone for handling all the status reports, and please let me know if there's anything you'd like me to add to this PR :+1: :+1:

pauamma commented on website/content/en/status/report-2023-04-2023-06/mfsbsd-building-integration.adoc:
```diff
@@ -0,0 +1,13 @@
+Contact: Soobin Rho <soobinrho@FreeBSD.org>
+
+This project creates an additional target of the weekly snapshots of -current and -stable versions of mfsBSD images in the src/release makefile.
+The problem this project aims to address is that, since only the release versions of mfsBSD images are produced currently, they tend to get out of sync with the tools in base.
```
```suggestion
Currently, only the release versions of mfsBSD images are produced.
This means they tend to get out of sync with the tools in base.
This project aims to address that problem.
```

Better flow by splitting it into 3 sentences.

soobinrho replied:
Very nice change! Thank you for the suggestion!

pauamma reviewed (commented):
Other than that, LGTM.

lsalvadore reviewed (changes requested):
Thanks for your report.

I suggest you add some context. I am unsure that all of our readers know about mfsBSD, I did not for example.

You could add one or two sentences to give a general definition and then point to a link (that you can also put in the links header) for readers wanting more details.

soobinrho commented:
> Thanks for your report.
> 
> I suggest you add some context. I am unsure that all of our readers know about mfsBSD, I did not for example.
> 
> You could add one or two sentences to give a general definition and then point to a link (that you can also put in the links header) for readers wanting more details.

Completely agreed. I'll make additional commit containing an explanation on what is mfsBSD. Thank you, @lsalvadore for the suggestion!

soobinrho commented:
I rebased this PR branch based on the upstream/main branch since there's been a few new commits.

```bash
git rebase https://github.com/freebsd/freebsd-doc.git
git remote add upstream https://github.com/freebsd/freebsd-doc.git
git fetch upstream
git rebase upstream/main
git push -f
```

lsalvadore commented on website/content/en/status/report-2023-04-2023-06/mfsbsd-building-integration.adoc:
```diff
@@ -0,0 +1,30 @@
+This means they tend to get out of sync with the tools in base.
+This project aims to address that problem.
+
+==== What is mfsBSD?
```
I would move this section up: I think it makes more sense to first define mfsBSD, then talk about your mfsBSD project.

lsalvadore commented on website/content/en/status/report-2023-04-2023-06/mfsbsd-building-integration.adoc:
```diff
@@ -0,0 +1,30 @@
+
+"mfsBSD is a toolset to create small-sized but full-featured mfsroot based distributions of FreeBSD that store all files in memory (MFS) [Memory File System] and load from hard drive, usb storage device or optical media.
+It can be used for a variety of purposes, including diskless systems, recovery partitions and remotely overwriting other operating systems."
+(link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[])
```
We generally avoid putting links out of sentences (unless it is in a links list).

I suggest to transform it in `link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[mfsBSD]`, using the first occurence of the msfBSD term in the body of your report.

lsalvadore commented on website/content/en/status/report-2023-04-2023-06/mfsbsd-building-integration.adoc:
```diff
@@ -0,0 +1,30 @@
+It can be used for a variety of purposes, including diskless systems, recovery partitions and remotely overwriting other operating systems."
+(link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[])
+
+Martin Matuska <mm@FreeBSD.org> is both the author of the white paper above and the maintainer of the mfsBSD repository: link:https://github.com/mmatuska/mfsbsd[]
```
Please use the `mailto` macro here:

`mailto:mm@FreeBSD[Martin Matuska]`

And use the link macro as following:

`link:https://github.com/mmatuska/mfsbsd[mfsBSD repository]`.

lsalvadore commented on website/content/en/status/report-2023-04-2023-06/mfsbsd-building-integration.adoc:
```diff
@@ -0,0 +1,30 @@
+
+==== Location of this project
+
+This project is a GSoC 2023 (Google Summer of Code) project, under mentorship from Juraj Lutter <otis@FreeBSD.org> and Joseph Mingrone <jrm@FreeBSD.org>
```
Please use the `mailto` macro here as well.

lsalvadore commented on website/content/en/status/report-2023-04-2023-06/mfsbsd-building-integration.adoc:
```diff
@@ -0,0 +1,30 @@
+This project is a GSoC 2023 (Google Summer of Code) project, under mentorship from Juraj Lutter <otis@FreeBSD.org> and Joseph Mingrone <jrm@FreeBSD.org>
+
+As a GSoC project, the official coding period is between May 29, 2023 and August 28, 2023.
+As a humble beginner in the open-source community, the author of this project welcomes all comments / suggestions / PR's to the project repository, which is where all the code for this project will be located for the duration of the official GSoC coding period: link:https://github.com/soobinrho/freebsd-src/tree/integrate-mfsBSD-building[]
```
`link:https://github.com/soobinrho/freebsd-src/tree/integrate-mfs-building[project repository]`

soobinrho replied:
Awesome! I'll make corresponding changes when I get back home. Thank you for these suggestions; they are awesome changes for my status report!

lsalvadore reviewed (changes requested):
A few minor changes and this is ready.

lsalvadore commented:
@sergio-carlavilla , @dbaio: can I proceed with the commit once this is ready? do you mentor approve?

Thanks.

soobinrho commented:
Thanks @lsalvadore again for the suggestions! I've implemented all the suggestions. Also, as suggested by my mentors @Jehops and @jurajlutter, I've squashed all the previous commits into one commit 👍 👍

Jehops commented on website/content/en/status/report-2023-04-2023-06/mfsbsd-building-integration.adoc:
```diff
@@ -0,0 +1,28 @@
+"link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[mfsBSD] is a toolset to create small-sized but full-featured mfsroot based distributions of FreeBSD that store all files in memory (MFS) [Memory File System] and load from hard drive, usb storage device or optical media.
+It can be used for a variety of purposes, including diskless systems, recovery partitions and remotely overwriting other operating systems."
+
+mailto:mm@FreeBSD.org[Martin Matuska] is both the author of the white paper above and the maintainer of the link:https://github.com/mmatuska/mfsbsd[mfsBSD repository].
```
How about this?  ...is both the author of the link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[white paper] and the...

I know you already link to the white paper above, but it took me a moment to realize what you meant by saying the "white paper *above*".  Another option could be to say "is both the author of the white paper linked above...".

Jehops replied:
Other than this optional change it looks good to me.

soobinrho replied:
Absolutely! That makes the writing much nicer. Changed to `link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[white paper]` now. Thanks @Jehops!

soobinrho commented:
All of this would not have been possible without the creator of mfsBSD. Thank you @mmatuska for creating and maintaining mfsBSD!

lsalvadore reviewed (approved):
Approved.

@sergio-carlavilla @dbaio : Do you mentor approve the commit?

@soobinrho : I am a mentored doc committer, I need to get mentor approval before committing your report to the doc repository.

dbaio commented:
> Approved.
> 
> @sergio-carlavilla @dbaio : Do you mentor approve the commit?

Yes, please, go ahead.
